### PR TITLE
Add DateDiff() and .Now for DateTimeOffset

### PR DESCRIPTION
**Pull request by viceroypenguin**
Limited to ms sql for now. When I set up new DB servers (PostgreSQL, Oracle, MySQL, MariaDB, etc.), I'll expand options

### CODE REVIEW

**Comment by sdanyliv** [comment]
Looks like for DB2 it is `CURRENT TIMESTAMP WITH TIME ZONE`

**Comment by MaceWindu** [comment]
https://www.ibm.com/support/producthub/db2/docs/content/SSEPGG_11.5.0/com.ibm.db2.luw.sql.ref.doc/doc/r0005886.html

**Comment by viceroypenguin** [comment]
@sdanyliv https://www.ibm.com/support/knowledgecenter/en/SSEPEK_11.0.0/sqlref/src/tpc/db2z_currenttimestamp.html says `SYSTIMESTAMP` is shortcut for `CURRENT TIMESTAMP WITH TIME ZONE`

**Comment by MaceWindu** [comment]
Note that your link is for DB2 z/OS, but we test against DB2 LUW

**Comment by viceroypenguin** [comment]
Ah. Good to know. Interesting.

**Comment by viceroypenguin** [comment]
Well. `CURRENT TIMESTAMP WITH TIME ZONE` was not accepted by DB2. So there's that. Also, apparently `SYSTIMESTAMP()` on Oracle doesn't fit into a `DateTimeOffset`? Back to the research center.

**Comment by viceroypenguin** [comment]
Removed DB2 - Doesn't like SYSTIMESTAMP, CURRENT TIME WITH TIME ZONE, or SYSTIMESTAMP(). If someone needs it later, it can be added.

**Comment by viceroypenguin** [comment]
@sdanyliv @MaceWindu  any reason not to pull this in 3.1.3?

File: Tests/Linq/Linq/DateTimeOffsetTests.cs [hunk 1]
```diff
@@ -480,7 +483,7 @@ public void GroupByDateTimeOffsetByAddSecondsTest([IncludeDataSources(true, Test
 		}
 
 		[Test]
-		public void GroupByDateTimeOffsetByAddYearsTest([IncludeDataSources(true, TestProvName.AllSqlServer2008Plus)] string context)
```
Comment (MaceWindu) [inline Tests/Linq/Linq/DateTimeOffsetTests.cs:483]: Why disable remote context?

**Review by MaceWindu — APPROVED** [review]

File: Tests/Linq/Linq/DateTimeOffsetTests.cs (hunk 1 above)
Comment (viceroypenguin) [reply Tests/Linq/Linq/DateTimeOffsetTests.cs:483]: Dropped when playing around with getting other dbs working. Will add back shortly.

**Pull request merged**